### ISSUE TITLE
Consider sending timing header from actuator

### ISSUE THREAD

**Issue by philwebb** [spring-projects/spring-boot#8736]
As suggested [in this tweet](https://twitter.com/taseroth/status/845316643324612609) it's worth looking to see if we can send the HTTP headers that are [automatically picked up by Chrome Devtools](https://ma.ttias.be/server-timings-chrome-devtools/).

**Comment by bclozel** [comment]
I don't think this feature would be useful for actuators - this would just give you how long each healthindicator took - something already considered in #9333.

Looking at the feature, I think this is meant to provide information about how time was spent on the server while processing an application request - some kind of application profiling feature that could be enabled with devtools. See an example of that with [the Symfony profiler](http://symfony.com/blog/new-in-symfony-3-1-web-debug-toolbar-and-profiler-enhancements).

This could be really useful, but I don't think we have the required hooks/instrumentation infrastructure to get those numbers for any application request — or at least this is a much larger endeavor than exposing those metrics.

Maybe we could expose spring-cloud-sleuth traces? This would only consider remote calls, though.

**Comment by bclozel** [comment]
See [SPR-17035](https://jira.spring.io/browse/SPR-17035).

**Comment by bclozel** [comment]
It would be interesting to see how https://github.com/sercasti/spring-httpserver-timings-java17 evolves, how the community is using it and if additional infrastructure is needed in Spring Boot or Micrometer to support it.

**Comment by bclozel** [comment]
The Framework issue was closed because of lack of demand. The `spring-httpserver-timings-java17` did not get much traction.

Meanwhile, tracing support in Spring has been improved dramatically and is more widely adopted compared to timing headers. I think #44431 would be a better fit in this space. I'm declining this issue as a result.
